monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A collection of transforms for signal operations
+https://github.com/Project-MONAI/MONAI/wiki/MONAI_Design
+"""
+
+from typing import Any, List, Optional
+
+import numpy as np
+
+from monai.transforms.transform import RandomizableTransform, Transform
+from monai.utils import optional_import
+from monai.utils.enums import TransformBackends
+
+zoom, has_zoom = optional_import("scipy.ndimage", name="zoom")
+resample_poly, has_resample_poly = optional_import("scipy.signal", name="resample_poly")
+fft, has_resample_fft = optional_import("scipy.signal", name="resample")
+shift, has_shift = optional_import("scipy.ndimage.interpolation", name="shift")
+square, has_square = optional_import("scipy.signal", name="square")
+iirnotch, has_iirnotch = optional_import("scipy.signal", name="iirnotch")
+filtfilt, has_filtfilt = optional_import("scipy.signal", name="filtfilt")
+spectrogram, has_spectrogram = optional_import("scipy.signal", name="spectrogram")
+central_frequency, has_central_frequency = optional_import("pywt", name="central_frequency")
+cwt, has_cwt = optional_import("pywt", name="cwt")
```

Comment:
Hi @rijobro ,

As you mentioned in the comments, I agree MONAI already supports many operations.
I think @doursand may be not familiar with what exists in MONAI.
Could you please help confirm the optional imports here and let's only keep the "must-have"?
For example, we can use MONAI transforms for zoom, torch API for fft?

Thanks in advance.

Reply:
There are fft utils in monai, not sure whether they are useful here. https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/fft_utils.py

Reply:
@Nic-Ma indeed I just realized that Zoom does the same as SignalResample, so I m removing it from my code ;-)
For numpy backend, it was done on purpose , I wanted to have something working initially on numpy and them update to torch as well.

@Can-Zhao Thanks for the tip, indeed I don't know well enough monai and all the stuffs which are already there, so I will try to use that one first